src/ros_vrpn_client.cpp

```diff
@@ -234,25 +294,18 @@ int main(int argc, char* argv[])
   private_nh.param<std::string>("vrpn_server_ip", vrpn_server_ip, std::string());
   private_nh.param<int>("vrpn_port", vrpn_port, 3883);
   private_nh.param<std::string>("vrpn_coordinate_system", coordinate_system_string, "vicon");
-  private_nh.param<std::string>("object_name", object_name, "bluebird");
+  private_nh.param<std::string>("object_name", object_name, "auk");
 
+  // Debug output
   std::cout << "vrpn_server_ip:" << vrpn_server_ip << std::endl;
   std::cout << "vrpn_port:" << vrpn_port << std::endl;
   std::cout << "vrpn_coordinate_system:" << coordinate_system_string << std::endl;
   std::cout << "object_name:" << object_name << std::endl;
 
-  if (coordinate_system_string == std::string("vicon"))
-  {
-
-  }
-  else if (coordinate_system_string == std::string("optitrack"))
-  {
-
-  }
-  else
-  {
-    ROS_FATAL("ROS param vrpn_coordinate_system should be either 'vicon' or 'optitrack'!");
-  }
+  // Checking for valid coordinate specification
+  CHECK((coordinate_system_string == std::string("vicon"))
```

Comment:
I think I may prefer `coordinate_system_string.compare("blah") == 0`, but whichever you think is more readable. (No need to do the std::string() cast with compare though).
http://www.cplusplus.com/reference/string/string/compare/